kernel/mmu.c

```diff
@@ -932,6 +932,28 @@ void z_mem_manage_boot_finish(void)
 #endif
 }
 
+int k_virt_free_range_max_get(void)
+{
+	int bit_val;
+	int free_pages = 0;
+	int free_pages_max = 0;
+	k_spinlock_key_t key = k_spin_lock(&z_mm_lock);
+
+	for (size_t i = 0; i < virt_region_bitmap.num_bits; ++i) {
+		sys_bitarray_test_bit(&virt_region_bitmap, i, &bit_val);
+		if (!bit_val) {
+			++free_pages;
+			continue;
+		}
+		free_pages_max = free_pages > free_pages_max ? free_pages : free_pages_max;
+		free_pages = 0;
```

Comment:
It seems like you are trying to return the size of the largest continuous block of virtual memory. If so, the function description in the header file needs to be clarified. And the name of the function needs to convey this too.

Reply:
Ok I will use k_mem_virt_free_get() as you have suggested